docs/contracts.rst

```diff
@@ -1406,24 +1406,24 @@ Using For
 
 The directive ``using A for B;`` can be used to attach library
 functions (from the library ``A``) to any type (``B``).
-These functions will receive the object they are called on
-as their first parameter (like the ``self`` variable in
-Python).
 
 The effect of ``using A for *;`` is that the functions from
 the library ``A`` are attached to any type.
 
-In both situations, all functions, even those where the
-type of the first parameter does not match the type of
-the object, are attached. The type is checked at the
+In both situations, all functions in the library are attached,
+even those where the type of the first parameter does not
```

Comment:
To be meaningful, this  needs the sentence about what attaching means (the one you deleted above).

Reply:
It's not deleted, just moved a paragraph or so lower down so that each fact stands alone for clarity. If you want it moved back I'll do so, but I think it's clearer this way.